lib/mosaic-routing/src/main/java/org/eclipse/mosaic/lib/routing/graphhopper/util/WayTypeEncoder.java

```diff
@@ -15,13 +15,19 @@
 
 package org.eclipse.mosaic.lib.routing.graphhopper.util;
 
+import org.eclipse.mosaic.lib.database.road.Way;
+
 import com.google.common.collect.BiMap;
 import com.google.common.collect.HashBiMap;
 import com.google.common.collect.Sets;
+import com.graphhopper.routing.ev.EdgeIntAccess;
+import com.graphhopper.routing.ev.IntEncodedValueImpl;
 
 import java.util.Set;
 
-public class WayTypeEncoder {
+public class WayTypeEncoder extends IntEncodedValueImpl {
```

Comment:
add javadoc

Reply:
done, and also fixed. I added a test and found that the `TYPE_MASK` field was wrong. I also found why I did reduced the numbers for the bitshifting. we only store the waytype as an integer in the first 8 bits of the integer, the rest is used to store the flags bitwise, here we can already start with the 9th bit (`1 << 8`).